### DISCUSSION TITLE
How to hide a sector?

### DISCUSSION THREAD

**Discussion by JoanCodinachUoc** [GrapesJS/grapesjs#5476] [Show and tell]
I want in the image components to hide the dimension sector. I try to modify the hide boolean in attributes but it doesn't change. The only way to dissapear the sector is to call removeSector but i then i have to add the sector again and doesn't works. Theres is a easy way to hide the sectors ? 

```
  if (block.attributes.blockId === 'image') {
     
          this.editor.StyleManager.removeSector('dimension');
        } else {
          this.editor.StyleManager.addSector('dimension', { name: 'dimension', open: true, properties: this.b });
        }
```
I did something like that but i supose that exist an easy way. Furthemore, the sector is created but not good because appears the fields outside the sector.

Thanks

**Comment by mosh-tudor** [comment]
@JoanCodinachUoc 
Have you tried using the `unstylable` attribute in your component?
```js
unstylable: ['width', 'height', 'max-width', 'min-height', 'margin', 'padding']
```

https://grapesjs.com/docs/api/component.html#component

**Reply by artf** [reply]
This is one way to control the StyleManager properties visibility directly from the component. If all properties of the sector are unstylable, the sector won't be shown.
Alternatively, here is how you would hide the sector directly via API:
```js
const sector = editor.Styles.getSector('dimension');
sector.set({ visible: false }):
```
